internal/collector/postgres.go

```diff
@@ -272,3 +285,56 @@ func EnablePostgresLogging(
 		}
 	}
 }
+
+// generateLogFilenameAndRotationAge takes a retentionPeriod and returns a
+// log_filename and log_rotation_age to be used to configure postgres logging
+func generateLogFilenameAndRotationAge(
+	retentionPeriod metav1.Duration,
+) (logFilename, logRotationAge string) {
+	number, interval := parseDurationForLogrotate(retentionPeriod)
```

Comment:
🤔 I wonder if translating the retention period twice is worth it. Would the logic below be easier if we went straight off a duration?

```go
hours := math.Ceil(retentionPeriod.Hours())

switch true {
case hours <= 1:	… // minutes
case hours <= 24:	… // hours
case hours <= 24 * 7:	… // weekdays
case hours <= 24 * 28:	… // days of month
default:	… // days of year
}
```